cyclonedx-ruby.gemspec

```diff
@@ -1,26 +1,79 @@
 # frozen_string_literal: true
 
+require_relative "lib/cyclonedx/ruby/version"
+
 Gem::Specification.new do |spec|
-  spec.name        = 'cyclonedx-ruby'
-  spec.version     = '1.2.0'
-  spec.date        = '2023-07-14'
+  spec.name = 'cyclonedx-ruby'
+  spec.version = Cyclonedx::Ruby::VERSION
+  spec.authors = ['Joseph Kobti', 'Steve Springett']
+  spec.email = ['josephkobti@outlook.com']
+
   spec.summary     = 'CycloneDX software bill-of-material (SBoM) generation utility'
   spec.description = 'CycloneDX is a lightweight software bill-of-material (SBOM) specification designed for use in application security contexts and supply chain component analysis. This Gem generates CycloneDX BOMs from Ruby projects.'
-  spec.authors     = ['Joseph Kobti', 'Steve Springett']
-  spec.email       = 'josephkobti@outlook.com'
   spec.homepage    = 'https://github.com/CycloneDX/cyclonedx-ruby-gem'
   spec.license     = 'Apache-2.0'
-
   spec.required_ruby_version = '>= 2.7.0'
 
-  spec.files = Dir.chdir(__dir__) do
-    `git ls-files -z`.split("\x0").reject do |f|
-      (File.expand_path(f) == __FILE__) || f.start_with?(*%w[bin/ test/ spec/ features/ .git .circleci appveyor])
+  # Linux distros often package gems and securely certify them independent
+  #   of the official RubyGem certification process. Allowed via ENV["SKIP_GEM_SIGNING"]
+  # Ref: https://gitlab.com/ruby-oauth/version_gem/-/issues/3
+  # Hence, only enable signing if `SKIP_GEM_SIGNING` is not set in ENV.
+  # See CONTRIBUTING.md
+  unless ENV.include?("SKIP_GEM_SIGNING")
+    user_cert = "certs/#{ENV.fetch("GEM_CERT_USER", ENV["USER"])}.pem"
+    cert_file_path = File.join(__dir__, user_cert)
+    cert_chain = cert_file_path.split(",")
+    cert_chain.select! { |fp| File.exist?(fp) }
+    if cert_file_path && cert_chain.any?
+      spec.cert_chain = cert_chain
+      if $PROGRAM_NAME.end_with?("gem") && ARGV[0] == "build"
+        spec.signing_key = File.join(Gem.user_home, ".ssh", "gem-private_key.pem")
+      end
     end
   end
-  spec.bindir = 'exe'
-  spec.executables = spec.files.grep(%r{\Aexe/}) { |f| File.basename(f) }
+
+  spec.metadata["homepage_uri"] = "https://#{spec.name.tr('_', '-')}.galtzo.com/"
+  spec.metadata["source_code_uri"] = "#{spec.homepage}/tree/v#{spec.version}"
+  spec.metadata["changelog_uri"] = "#{spec.homepage}/blob/v#{spec.version}/CHANGELOG.md"
+  spec.metadata["bug_tracker_uri"] = "#{spec.homepage}/issues"
+  spec.metadata["documentation_uri"] = "https://www.rubydoc.info/gems/#{spec.name}/#{spec.version}"
```

Comment:
please remove, as long as we dong host any docs there.

Reply:
They are actually already hosted there, but the documentation isn't great in the current release. The next release will have excellent YARD documentation.

https://rubydoc.info/gems/cyclonedx-ruby